### PR TITLE
Handle NoHttpResponseException

### PR DESCRIPTION
**Pull request by okal**
By throwing a more user-friendly exception.

### CODE REVIEW

**Comment by pld** [comment]
@okal instead of having the add and revert comment you can remove both commits (bde and c7d) from the tree using `rebase -i`

File: src/milia/api/io.clj [hunk 1]
```diff
@@ -123,19 +124,26 @@
...
-          (do
-            (refresh-temp-token)
-            (req-fn)))))
-     ;; To avoid NPE, default to a fake 555 status if no status is returned
-     (catch #(<= 400 (:status % 555)) response response))))
+          (refresh-temp-token)
+          (send-request))
+        (catch NoHttpResponseException _
+          ;; Because Core doesn't respond with a 401 on unauthorized PATCH requests
+          (refresh-temp-token)
```
Comment (Santei) [inline src/milia/api/io.clj]: Why did you remove the `do` here? We want to refresh-temp-token then send-request.

Comment (okal) [inline src/milia/api/io.clj]: It wasn't necessary. Both things still happen in sequence.

Comment (Santei) [inline src/milia/api/io.clj]: Alright. That's fine. Was just worried about the sequence.

**Comment by Santei** [comment]
Looks okay. Hope this will finally close https://github.com/onaio/zebra/issues/2845.
